test/TesterInternal/MessageScheduling/ReentrancyTests.cs

```diff
@@ -99,20 +100,18 @@ public void NonReentrantGrain_WithMayInterleavePredicate_StreamItemDelivery_When
         }
 
         [Fact, TestCategory("Functional"), TestCategory("Tasks"), TestCategory("Reentrancy")]
-        public void NonReentrantGrain_WithMayInterleavePredicate_WhenPredicateThrows()
+        public async Task NonReentrantGrain_WithMayInterleavePredicate_WhenPredicateThrows()
         {
             var grain = this.fixture.GrainFactory.GetGrain<IMayInterleavePredicateGrain>(GetRandomGrainId());
             grain.SetSelf(grain).Wait();
             try
             {
-                grain.Exceptional().Wait(2000);
+                await grain.Exceptional().WithTimeout(TimeSpan.FromSeconds(2));
             }
             catch (Exception ex)
             {
-                Assert.IsType<OrleansException>(ex.GetBaseException());
-                Assert.NotNull(ex.GetBaseException().InnerException);
-                Assert.IsType<ApplicationException>(ex.GetBaseException().InnerException);
-                Assert.True(ex.GetBaseException().InnerException?.Message == "boom",
+                Assert.IsType<ApplicationException>(ex);
+                Assert.True(ex.Message == "boom",
```

Comment:
I think it's better not to wrap the exception in an OrleansException (since it isn't one), but to instead throw it as-is. Hence the test needed updating.